localstack/utils/docker.py

```diff
@@ -596,5 +803,559 @@ def mountable_tmp_file():
         TMP_FILES.append(f)
         return f
 
+    @staticmethod
+    def append_without_latest(image_names):
+        suffix = ":latest"
+        for image in list(image_names):
+            if image.endswith(suffix):
+                image_names.append(image[: -len(suffix)])
+
+    @staticmethod
+    def tar_path(path, target_path, is_dir: bool):
+        f = tempfile.NamedTemporaryFile()
+        with tarfile.open(mode="w", fileobj=f) as t:
+            abs_path = os.path.abspath(path)
+            arcname = (
+                os.path.basename(path)
+                if is_dir
+                else (os.path.basename(target_path) or os.path.basename(path))
+            )
+            t.add(abs_path, arcname=arcname)
+
+        f.seek(0)
+        return f
+
+    @staticmethod
+    def untar_to_path(tardata, target_path):
+        target_path = Path(target_path)
+        with tempfile.NamedTemporaryFile() as dest:
+            for d in tardata:
+                dest.write(d)
+            dest.seek(0)
+            with tarfile.open(mode="r", fileobj=dest) as t:
```

Comment:
Wondering if we need the NamedTemporaryFile here. Could this also work if we use `io.BytesIO` to wrap the `tardata` bytes into a file-like object?

```
with tarfile.open(mode="r", fileobj=io.BytesIO(tardata)) as t:
    ...
```

Reply:
Tardata is a generator not a bytes object, but `io.BytesIO(b"".join(b for b in tardata)))` should work